src/Plugin/Action/SystemPageRedirect.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\rules\Plugin\Action\SystemPageRedirect.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Drupal\Core\Routing\RedirectDestinationInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Provides "Page redirect" rules action.
+ *
+ * @Action(
+ *   id = "rules_page_redirect",
+ *   label = @Translation("Page redirect"),
+ *   category = @Translation("System"),
+ *   context = {
+ *     "url" = @ContextDefinition("uri",
```

Comment:
I think this should be "string" since a Drupal path is not a valid uri, but can still be used here.